racket/src/ChezScheme/c/segment.c

```diff
@@ -568,18 +575,154 @@ static void contract_segment_table(uptr base, uptr end) {
    thread-specific, the bracketing functions disable execution of the
    code's memory while enabling writing.
 
-   Note that these function will not work for a W^X implementation
-   where each page's disposition is process-wide. Indeed, a
-   process-wide W^X disposition seems incompatible with the Chez
+   A process-wide W^X disposition seems incompatible with the Chez
    Scheme rule that a foreign thread is allowed to invoke a callback
    (as long as the callback is immobile/locked) at any time --- even,
    say, while Scheme is collecting garbage and needs to write to
-   executable pages. */
+   executable pages.  However, on platforms where such a disposition
+   is enforced (eg. iOS), we provide a best-effort implementation that
+   flips pages between W and X for the minimal set of generations
+   possible (depending on the context) in an effort to minimize the
+   chances of a page being flipped while a thread is executing code
+   off of it.
+*/
 
-void S_thread_start_code_write(void) {
+void S_thread_start_code_write(ptr tc, IGEN maxg, IBOOL current, void *hint) {
+#if defined(WRITE_XOR_EXECUTE_CODE)
+  enable_code_write(tc, maxg, 1, current, hint);
+#else
+  (void)tc;
```

Comment:
Normally, an `UNUSED` annotation is written on the function argument to declare unused variables. In this case, of course, it's only unused when `WRITE_XOR_EXECUTE_CODE`, so I'd define a new `UNUSED_EXCEPT_W_XOR_X` (or something with a better name) and use that.